### PR TITLE
Prefs [Task][27972] Handle event click for News screen

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task_27972: Handle event click for News screen](https://edu-redmine.sun-asterisk.vn/issues/27972)

## WHAT
- Fix some bug
- Check connection internet
- Display webview when click item

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Trainer reivew (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt [hunk 1]
```diff
@@ -39,7 +41,11 @@ class DetailCountriesFragment :
         initAdapter()
         initPresenter()
         initDialog()
-        presenter?.start()
+        if (isConnection) {
+            presenter?.start()
+        } else {
+            noInternetConnection()
```
Comment (tuanpa-0908) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: Tên hàm là động từ

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt [hunk 2]
```diff
@@ -77,11 +83,15 @@ class DetailCountriesFragment :
...
+        if (isConnection) {
+            when (view.id) {
+                R.id.buttonSearch -> searchCountries()
+                R.id.textViewInfected -> sortCountries(TOTAL_CONFIRMED)
+                R.id.textViewDeath -> sortCountries(TOTAL_DEATHS)
+                R.id.textViewRecovered -> sortCountries(TOTAL_RECOVERED)
+            }
+        } else {
+            connectInternetFail()
         }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: viết bỏ else tránh code thụt đầu vào quá nhiều
if(!isConnection) {
    notifyNoNetworkConnection()
    return
}

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt [hunk 3]
```diff
@@ -116,8 +127,13 @@ class DetailCountriesFragment :
 
     private fun searchCountries() {
         val value = editTextSearch.text.toString()
-        presenter?.searchCountries(value)
+        if (isConnection) {
+            presenter?.searchCountries(value)
+        } else {
+            connectInternetFail()
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: if(!isConnection) {
    notifyNoNetworkConnection()
    return
}

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt [hunk 4]
```diff
@@ -184,6 +200,8 @@ class DetailCountriesFragment :
         updateIcon()
     }
 
+    private fun connectInternetFail() = showMessage(getString(R.string.msg_connection_fail))
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]:
```suggestion
    private fun notify....() = showMessage(getString(R.string.msg_connection_fail))
```

Comment (huytq-2422) [reply app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: Dùng snackBar

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/web/WebNewsActivity.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        webViewNews.webViewClient = WebViewClient()
+        webViewNews.loadUrl(url)
+    }
+
+    companion object {
+        private const val EXTRA_LINK_WEB = "EXTRA_LINK_WEB"
+        fun getIntent(context: Context, link: String): Intent {
+            val intent = Intent(context, WebNewsActivity::class.java)
+            intent.putExtra(EXTRA_LINK_WEB, link)
+            context.startActivity(intent)
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/web/WebNewsActivity.kt]: sao lại start luôn ở đây -_-

Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/web/WebNewsActivity.kt]:
```suggestion
        fun getIntent(context: Context, link: String): Intent =
            Intent(context, WebNewsActivity::class.java)
                .putExtra(EXTRA_LINK_WEB, link)
```

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt (hunk 2 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: Oki

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/web/WebNewsActivity.kt (hunk 5 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/ui/web/WebNewsActivity.kt]: Lỗi kỹ thuật ^_^

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/NetworkUtil.kt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val networkCapabilities = connectivityManager.activeNetwork ?: return false
+        val actNetwork =
+            connectivityManager.getNetworkCapabilities(networkCapabilities) ?: return false
+        return when {
+            actNetwork.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) -> true
+            actNetwork.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) -> true
+            actNetwork.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET) -> true
+            else -> false
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/utils/NetworkUtil.kt]:
```suggestion
        return condition1 or condition2 or condition3
```

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt [hunk 7]
```diff
@@ -39,7 +41,11 @@ class DetailCountriesFragment :
         initAdapter()
         initPresenter()
         initDialog()
-        presenter?.start()
+        if (isConnection) {
+            presenter?.start()
+        } else {
+            connectInternetFail()
```
Comment (tuanpa-0908) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt:47]: notifyConnectFailed() or notifyConnectInternetFailed() . connectInternetFail() vẫn không phải động từ mà :v

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt (hunk 4 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: Oki

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/NetworkUtil.kt [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    fun isConnection(context: Context): Boolean {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val networkCapabilities = connectivityManager.activeNetwork ?: return false
+        val actNetwork =
+            connectivityManager.getNetworkCapabilities(networkCapabilities) ?: return false
+        return actNetwork.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) or
+                actNetwork.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) or
+                actNetwork.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET) or
+                false
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/utils/NetworkUtil.kt]: có cần or false nữa không nhỉ?

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt [hunk 9]
```diff
@@ -39,6 +38,10 @@ class DetailCountriesFragment :
         initAdapter()
         initPresenter()
         initDialog()
+        if (!isConnection) {
+            context?.showToast(getString(R.string.msg_connection_fail))
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: máy thằng này cũng showSnackBar

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/NetworkUtil.kt (hunk 8 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/utils/NetworkUtil.kt]: Thừa huy ạ

File: app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt (hunk 9 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/ui/detail/DetailCountriesFragment.kt]: À oki

**Review by aug16vcc — APPROVED** [review]

**Pull request merged**